orion-server/src/main/java/com/pinterest/orion/core/actions/kafka/ClusterRecoveryAction.java

```diff
@@ -0,0 +1,291 @@
+                    existingNode.getCurrentNodeInfo().getNodeId(),
+                    brokerId
+            );
+            // setting these attributes to indicate that the node doesn't exist in cluster map, and should skip any node-related checks
+            brokerRecoveryAction.setAttribute(BrokerRecoveryAction.ATTR_NODE_EXISTS_KEY, false);
+            brokerRecoveryAction.setAttribute(BrokerRecoveryAction.ATTR_NONEXISTENT_HOST_KEY, extractedName);
+        }
+        if (maybeDeadBrokers.contains(brokerId)) {
+            // Setting this flag in the action will restart the broker before replacing the broker
+            brokerRecoveryAction.setAttribute(BrokerRecoveryAction.ATTR_TRY_TO_RESTART_KEY, true);
+            String restartNote =
+                    "BrokerRecoveryAction will try to restart node " + brokerId + " before replacing it.";
+            logger.info(restartNote);
+            getResult().appendOut(restartNote);
+        }
+        String note = String.format(
+                "Trigger BrokerRecoveryAction for node %s in cluster %s. Check the child action for details.",
+                brokerId,
+                cluster.getClusterId());
+        logger.info(note);
+        getResult().appendOut(note);
+        dispatchChildAction(brokerRecoveryAction);
+    }
+
+    /**
+     * This method adds a child action to the action list.
+     * It also dispatches the child action to the action engine.
+     * @param childAction the child action to be added.
+     * @throws Exception if the child action is not triggered successfully.
+     */
+    private void dispatchChildAction(Action childAction) throws Exception {
+        this.getChildren().add(childAction);
+        getEngine().dispatchChild(this, childAction);
+    }
+
+    /**
+     * This method triggers a BrokerRecoveryAction for each broker in candidates.
+     * It checks if the number of candidates is more than one.
+     * If the number of candidates is more than one, it will alert and not trigger any action.
+     * @param candidates a set of broker ids that are in bad states.
+     * @return true if all the child actions are triggered successfully.
+     * @throws Exception if any of the child actions is not triggered successfully.
+     */
+    protected boolean healBrokers(Set<String> candidates) throws Exception {
+        String output;
+        boolean isActionTriggered = false;
+        if (candidates.size() == 1) {
+            output = String.format(
+                    "ClusterRecoveryAction attempts to recover broker %s. ",
+                    candidates.iterator().next());
+            String brokerId = candidates.iterator().next();
+            logger.info(output);
+            triggerBrokerRecoveryAction(brokerId);
+            isActionTriggered = true;
+        } else if (candidates.size() > 1){
+            // more than 1 brokers are dead... better alert and have human intervention
```

Comment:
Is this comment still accurate? If we're leaving the responsibility of handling multiple failures to another workflow, we should clarify it here.

Reply:
Yes. This part is still accurate. I add the logic of handling multi-failures in an internal class that will override this class. The ossorion function will perform as same as before. 

If you think I should move the logic to ossorion, I can do that.

Reply:
Add logic into ossorion